design-documents/separate-config.md

```diff
@@ -0,0 +1,34 @@
+### Overview
+
+Create new separate file called modules.php that contain
+```php 
+[
+    modules => [...]
+}
+```
+
+###Problem
+
+File config .php contains shared website/stores and shared configuration. 
+This file include in git index and check changes
+This helps when we need to share some configuration across developers. 
+
+The problem is when we install new module or change sequence, after command:
+
+```bash
+php bin/magento setup:upgrade
+```
+Magento changes __modules__ array and git marks this file as changed. 
```

Comment:
Because it should be changed. The workflow as I imagine it is:
1. Install new module on dev environment (`config.php` updated)
2. Push code to staging (new `config.php` is fetched with new list and sequence of modules) - the application can now be updated correctly
3. Push code to prod (same as in step 2)

Can you please describe in more details the situation where you have modules sequence being changed on dev environment, but it doesn't make sense to change it on prod?

Reply:
Hi @buskamuza 

I duplicate cases from my side

The situation from my side

Yes I know that env.php is merged with config.php and has priority.
I have a big project and 2 different teams(work remote). What I need do when I working or do code review?
a) Change branch.
b) Start command setup:upgrade, setup:di:compile, setup:static-content:deploy for chek that changes not broken deployment or migration scripts and functionality are working
c) optionality make some changes in branch
Then if I create commit I need manually remove config.php from commit because it contained config data for different websites and stores but I not needed commit data from modules.
If I change branch I need create stash because I can't change branch while config.php are changed.

Thank's